CRM/Core/Payment/OmnipayMultiProcessor.php

```diff
@@ -512,6 +577,8 @@ private function getCreditCardObjectParams($params) {
       $cardFields['billingState'] = CRM_Core_PseudoConstant::stateProvinceAbbreviation($cardFields['billingState']);
     }
 
+    $this->copyShippingFieldsFromBillingIfEmpty($cardFields);
```

Comment:
@elcapo At least with sagepay there are two ways to handle the missing shipping fields. Either copy them or set `billingForShipping` to 1 in the omnipay sagepay gateway params - see https://github.com/eileenmcnaughton/nz.co.fuzion.omnipaymultiprocessor/pull/42/files#diff-71ff8f521b77ef1a6c6643105f8b24d5R60

Reply:
I've just tried removing the `copyShippingFieldsFromBillingIfEmpty` function in favor of the `gateway_params` you suggest but that takes me again to the _3140 : The DeliveryCountry value is invalid_ error message. Also, I didn't find any usage of the `gateway_params` in the extension. Maybe it has been deprecated?

Reply:
It wasn't fully implemented in this extension yet - see https://github.com/eileenmcnaughton/nz.co.fuzion.omnipaymultiprocessor/pull/81

Reply:
Thanks! Good to know! I'll do that your way when your PR is accepted. It looks cleaner than mine.